### ISSUE TITLE
Redis 6.2.14 crashed by signal: 11, si_code: 2

### ISSUE THREAD

**Issue by mkoser** [RediSearch/RediSearch#4719]
Our organization has experienced this crash intermittently on our redis cluster in all environments in which it runs.

<details><summary>Crash Report</summary>

```
=== REDIS BUG REPORT START: Cut & paste starting from here ===
1411:S 05 Jun 2024 02:00:31.254 # Redis 6.2.14 crashed by signal: 11, si_code: 2
1411:S 05 Jun 2024 02:00:31.254 # Accessing address: 0x7fb30f95bd90
1411:S 05 Jun 2024 02:00:31.254 # Crashed running the instruction at: 0x7fb30f95bd90

------ STACK TRACE ------
EIP:
[0x7fb30f95bd90]

Backtrace:
/lib64/libpthread.so.0(+0x12cf0)[0x7fb4d3e35cf0]
/usr/bin/redis-server 0.0.0.0:6380 [cluster] /etc/redis/6380/redis.conf(invalidFunctionWasCalled+0x0)[0x55bd6e22cd40]
/usr/lib64/redis/modules/redisearch-coord.so(+0x1e75db)[0x7fb4c3f925db]
/usr/lib64/redis/modules/redisearch-coord.so(+0x1d8bff)[0x7fb4c3f83bff]
/lib64/libpthread.so.0(+0x81ca)[0x7fb4d3e2b1ca]
/lib64/libc.so.6(clone+0x43)[0x7fb4d3a97e73]

------ REGISTERS ------
1411:S 05 Jun 2024 02:00:31.254 # 
RAX:00007fb4aeda3600 RBX:00007fb44d505580
RCX:00007fb480000000 RDX:0000000000000000
RDI:00007fb4aeda3600 RSI:76746d6100000030
RBP:00007fb4ab7f5d30 RSP:00007fb4c3b7f328
R8 :00000000000084bf R9 :0000000000000050
R10:00007fb4c3b7ff28 R11:000055bd6e34b880
R12:00007fb4aeda3600 R13:0000000000000000
R14:00007fb4d1a40138 R15:00007fb4d1a0d940
RIP:00007fb30f95bd90 EFL:0000000000010206
CSGSFS:002b000000000033
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f337) -> 0000000000000000
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f336) -> 0000000000000000
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f335) -> 0000000000000000
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f334) -> 0000000000000000
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f333) -> 0000000000000000
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f332) -> 0000000000000000
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f331) -> 0000000000000000
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f330) -> 000000302d6c6f6f
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f32f) -> 702d646165726874
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f32e) -> 00007fb4c3f83bff
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f32d) -> 00007fb4d1a400e0
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f32c) -> 00007fb4d1a400c0
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f32b) -> 00007fb3f9e11088
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f32a) -> 00007fb4d1a0d968
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f329) -> 00007fb4d1a0d940
1411:S 05 Jun 2024 02:00:31.255 # (00007fb4c3b7f328) -> 00007fb4c3f925db

------ INFO OUTPUT ------
# Server
redis_version:6.2.14
redis_git_sha1:00000000
redis_git_dirty:0
redis_build_id:cdbda0ad900ddcdb
redis_mode:cluster
os:Linux 4.18.0-513.24.1.el8_9.x86_64 x86_64
arch_bits:64
monotonic_clock:POSIX clock_gettime
multiplexing_api:epoll
atomicvar_api:c11-builtin
gcc_version:8.5.0
process_id:1411
process_supervised:systemd
run_id:e4ce6550078b4c26d443a8979367770746c664f5
tcp_port:6380
server_time_usec:1717567231176266
uptime_in_seconds:1560580
uptime_in_days:18
hz:10
configured_hz:10
lru_clock:6291199
executable:/usr/bin/redis-server
config_file:/etc/redis/6380/redis.conf
io_threads_active:0

# Clients
connected_clients:36
cluster_connections:18
maxclients:10000
client_recent_max_input_buffer:40
client_recent_max_output_buffer:0
blocked_clients:0
tracking_clients:0
clients_in_timeout_table:0

# Memory
used_memory:6216006024
used_memory_human:5.79G
used_memory_rss:6785236992
used_memory_rss_human:6.32G
used_memory_peak:6579357752
used_memory_peak_human:6.13G
used_memory_peak_perc:94.48%
used_memory_overhead:67964840
used_memory_startup:1605256
used_memory_dataset:6148041184
used_memory_dataset_perc:98.93%
allocator_allocated:6216897952
allocator_active:6794174464
allocator_resident:6875643904
total_system_memory:50337116160
total_system_memory_human:46.88G
used_memory_lua:36864
used_memory_lua_human:36.00K
used_memory_scripts:880
used_memory_scripts_human:880B
number_of_cached_scripts:4
maxmemory:0
maxmemory_human:0B
maxmemory_policy:noeviction
allocator_frag_ratio:1.09
allocator_frag_bytes:577276512
allocator_rss_ratio:1.01
allocator_rss_bytes:81469440
rss_overhead_ratio:0.99
rss_overhead_bytes:-90406912
mem_fragmentation_ratio:1.09
mem_fragmentation_bytes:569253656
mem_not_counted_for_evict:0
mem_replication_backlog:1048576
mem_clients_slaves:0
mem_clients_normal:738056
mem_aof_buffer:0
mem_allocator:jemalloc-5.1.0
active_defrag_running:0
lazyfree_pending_objects:0
lazyfreed_objects:0

# Persistence
loading:0
current_cow_size:127275008
current_cow_size_age:4
current_fork_perc:50.64
current_save_keys_processed:411649
current_save_keys_total:812819
rdb_changes_since_last_save:107888
rdb_bgsave_in_progress:1
rdb_last_save_time:1717566415
rdb_last_bgsave_status:ok
rdb_last_bgsave_time_sec:56
rdb_current_bgsave_time_sec:30
rdb_last_cow_size:4825088
aof_enabled:0
aof_rewrite_in_progress:0
aof_rewrite_scheduled:0
aof_last_rewrite_time_sec:-1
aof_current_rewrite_time_sec:-1
aof_last_bgrewrite_status:ok
aof_last_write_status:ok
aof_last_cow_size:0
module_fork_in_progress:0
module_fork_last_cow_size:0

# Stats
total_connections_received:500308
total_commands_processed:92182523
instantaneous_ops_per_sec:4
total_net_input_bytes:253393131700
total_net_output_bytes:6044594357
instantaneous_input_kbps:0.02
instantaneous_output_kbps:0.46
rejected_connections:0
sync_full:0
sync_partial_ok:0
sync_partial_err:0
expired_keys:0
expired_stale_perc:0.00
expired_time_cap_reached_count:0
expire_cycle_cpu_milliseconds:0
evicted_keys:0
keyspace_hits:19604366
keyspace_misses:0
pubsub_channels:1
pubsub_patterns:0
latest_fork_usec:124859
total_forks:18818
migrate_cached_sockets:0
slave_expires_tracked_keys:0
active_defrag_hits:0
active_defrag_misses:0
active_defrag_key_hits:0
active_defrag_key_misses:0
tracking_total_keys:0
tracking_total_items:0
tracking_total_prefixes:0
unexpected_error_replies:0
total_error_replies:212199
dump_payload_sanitizations:0
total_reads_processed:64512176
total_writes_processed:5190452
io_threaded_reads_processed:0
io_threaded_writes_processed:0

# Replication
role:slave
master_host:10.19.50.181
master_port:6379
master_link_status:up
master_last_io_seconds_ago:4
master_sync_in_progress:0
slave_read_repl_offset:14506269563106
slave_repl_offset:14506269563106
slave_priority:100
slave_read_only:1
replica_announced:1
connected_slaves:0
master_failover_state:no-failover
master_replid:f43772840cbe0ad05ddd40b40e03a01f5284a5db
master_replid2:499f60e7c474aa34b1ec5b9d462a364ed2ba8189
master_repl_offset:14506269563106
second_repl_offset:14258276115088
repl_backlog_active:1
repl_backlog_size:1048576
repl_backlog_first_byte_offset:14506268514531
repl_backlog_histlen:1048576

# CPU
used_cpu_sys:10238.361810
used_cpu_user:8249.226562
used_cpu_sys_children:25503.884673
used_cpu_user_children:206041.814261
used_cpu_sys_main_thread:2482.284984
used_cpu_user_main_thread:157.502322

# Modules
module:name=ReJSON,ver=20606,api=1,filters=0,usedby=[search],using=[],options=[handle-io-errors]
module:name=search,ver=20801,api=1,filters=0,usedby=[],using=[ReJSON],options=[handle-io-errors]

# Commandstats
cmdstat_exec:calls=4097293,usec=147056815,usec_per_call=35.89,rejected_calls=0,failed_calls=0
cmdstat__FT._DROPINDEXIFX:calls=2170,usec=212904,usec_per_call=98.11,rejected_calls=0,failed_calls=0
cmdstat_zremrangebyscore:calls=4200,usec=110386,usec_per_call=26.28,rejected_calls=0,failed_calls=0
cmdstat_hmset:calls=17414012,usec=1449515566,usec_per_call=83.24,rejected_calls=0,failed_calls=0
cmdstat__FT.INFO:calls=572,usec=36561,usec_per_call=63.92,rejected_calls=0,failed_calls=1
cmdstat_del:calls=4683283,usec=110408708,usec_per_call=23.58,rejected_calls=0,failed_calls=0
cmdstat__FT.ALIASUPDATE:calls=2170,usec=39469,usec_per_call=18.19,rejected_calls=0,failed_calls=0
cmdstat__FT._CREATEIFNX:calls=2170,usec=915702,usec_per_call=421.98,rejected_calls=0,failed_calls=0
cmdstat_zincrby:calls=7816056,usec=104907568,usec_per_call=13.42,rejected_calls=0,failed_calls=0
cmdstat_rpush:calls=14398,usec=348535,usec_per_call=24.21,rejected_calls=0,failed_calls=0
cmdstat_cluster:calls=3592789,usec=99749136,usec_per_call=27.76,rejected_calls=351,failed_calls=0
cmdstat_latency:calls=416068,usec=1641217,usec_per_call=3.94,rejected_calls=0,failed_calls=208034
cmdstat_readonly:calls=165,usec=101,usec_per_call=0.61,rejected_calls=153,failed_calls=0
cmdstat_script:calls=402,usec=1182,usec_per_call=2.94,rejected_calls=0,failed_calls=0
cmdstat_expire:calls=3327520,usec=6574045,usec_per_call=1.98,rejected_calls=0,failed_calls=0
cmdstat__FT._LIST:calls=526,usec=6359,usec_per_call=12.09,rejected_calls=0,failed_calls=0
cmdstat_lpop:calls=14398,usec=512730,usec_per_call=35.61,rejected_calls=0,failed_calls=0
cmdstat_get:calls=0,usec=0,usec_per_call=0.00,rejected_calls=484,failed_calls=0
cmdstat_slowlog:calls=416068,usec=1448134,usec_per_call=3.48,rejected_calls=0,failed_calls=0
cmdstat_subscribe:calls=505,usec=1304,usec_per_call=2.58,rejected_calls=0,failed_calls=0
cmdstat_multi:calls=4097293,usec=650982,usec_per_call=0.16,rejected_calls=0,failed_calls=0
cmdstat_info:calls=619641,usec=35555531,usec_per_call=57.38,rejected_calls=0,failed_calls=0
cmdstat_zpopmin:calls=9236,usec=14197088,usec_per_call=1537.15,rejected_calls=0,failed_calls=0
cmdstat_set:calls=40434904,usec=366832887,usec_per_call=9.07,rejected_calls=0,failed_calls=0
cmdstat_auth:calls=474299,usec=2502170,usec_per_call=5.28,rejected_calls=0,failed_calls=0
cmdstat__FT.TAGVALS:calls=3,usec=209055,usec_per_call=69685.00,rejected_calls=0,failed_calls=0
cmdstat_rename:calls=976,usec=73762064,usec_per_call=75575.88,rejected_calls=0,failed_calls=0
cmdstat_zadd:calls=19053,usec=24476488,usec_per_call=1284.65,rejected_calls=0,failed_calls=0
cmdstat_search.CLUSTERREFRESH:calls=1559684,usec=138411754,usec_per_call=88.74,rejected_calls=0,failed_calls=0
cmdstat_select:calls=3,usec=4,usec_per_call=1.33,rejected_calls=0,failed_calls=0
cmdstat_client:calls=419035,usec=30532187,usec_per_call=72.86,rejected_calls=0,failed_calls=1978
cmdstat_sadd:calls=1,usec=47,usec_per_call=47.00,rejected_calls=0,failed_calls=0
cmdstat_hset:calls=33,usec=4947,usec_per_call=149.91,rejected_calls=0,failed_calls=0
cmdstat_ping:calls=772960,usec=1104171,usec_per_call=1.43,rejected_calls=4,failed_calls=0
cmdstat_hdel:calls=349528,usec=13730067,usec_per_call=39.28,rejected_calls=0,failed_calls=0
cmdstat_echo:calls=279,usec=123,usec_per_call=0.44,rejected_calls=710,failed_calls=0
cmdstat_config:calls=209054,usec=13540449,usec_per_call=64.77,rejected_calls=0,failed_calls=0
cmdstat_unlink:calls=1411776,usec=19501681,usec_per_call=13.81,rejected_calls=0,failed_calls=0

# Errorstats
errorstat_CLUSTERDOWN:count=345
errorstat_ERR:count=210609
errorstat_LOADING:count=1105
errorstat_MOVED:count=139
errorstat_Unknown:count=1

# Cluster
cluster_enabled:1

# Keyspace
db0:keys=812816,expires=636759,avg_ttl=0

------ CLIENT LIST OUTPUT ------
id=5329713 addr=10.19.35.26:40846 laddr=10.19.50.184:6380 fd=37 name=catalog-api-l-intnetcore206 age=66156 idle=39 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5329722 addr=10.19.35.23:51132 laddr=10.19.50.184:6380 fd=43 name=catalog-api-l-intnetcore203 age=66155 idle=1 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5329723 addr=10.19.35.23:51140 laddr=10.19.50.184:6380 fd=44 name=catalog-api-l-intnetcore203 age=66155 idle=54 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=5329714 addr=10.19.35.26:40862 laddr=10.19.50.184:6380 fd=40 name=catalog-api-l-intnetcore206 age=66156 idle=35 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=5330816 addr=10.19.35.35:46386 laddr=10.19.50.184:6380 fd=54 name=catalog-admin-api-l-intnetcore215 age=65850 idle=32 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5330817 addr=10.19.35.35:46392 laddr=10.19.50.184:6380 fd=57 name=catalog-admin-api-l-intnetcore215 age=65850 idle=30 flags=Pr db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=10946 addr=10.19.50.182:58934 laddr=10.19.50.184:6380 fd=73 name= age=1557694 idle=1557694 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=auth user=default redir=-1
id=8003 addr=10.19.50.185:53294 laddr=10.19.50.184:6380 fd=64 name= age=1558515 idle=1558515 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=auth user=default redir=-1
id=10945 addr=10.19.50.182:58918 laddr=10.19.50.184:6380 fd=35 name= age=1557694 idle=3630 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=_FT.INFO user=default redir=-1
id=5329957 addr=10.19.35.32:52940 laddr=10.19.50.184:6380 fd=32 name=catalog-admin-api-l-intnetcore212 age=66090 idle=48 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5329958 addr=10.19.35.32:52950 laddr=10.19.50.184:6380 fd=33 name=catalog-admin-api-l-intnetcore212 age=66090 idle=37 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=730 addr=10.19.50.184:33166 laddr=10.19.50.184:6380 fd=18 name= age=1560545 idle=31 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=_FT.INFO user=default redir=-1
id=5329400 addr=10.19.35.24:55982 laddr=10.19.50.184:6380 fd=17 name=catalog-api-l-intnetcore204 age=66240 idle=23 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5329401 addr=10.19.35.24:55992 laddr=10.19.50.184:6380 fd=19 name=catalog-api-l-intnetcore204 age=66240 idle=23 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=3763 addr=10.19.50.181:50482 laddr=10.19.50.184:6380 fd=92 name= age=1559694 idle=1559694 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=auth user=default redir=-1
id=4076961 addr=10.19.50.183:60782 laddr=10.19.50.184:6380 fd=61 name= age=417562 idle=417562 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=auth user=default redir=-1
id=727 addr=10.19.50.184:33156 laddr=10.19.50.184:6380 fd=15 name= age=1560545 idle=1560545 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=auth user=default redir=-1
id=5329513 addr=10.19.35.31:35734 laddr=10.19.50.184:6380 fd=46 name=catalog-admin-api-l-intnetcore211 age=66210 idle=35 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5329514 addr=10.19.35.31:35742 laddr=10.19.50.184:6380 fd=49 name=catalog-admin-api-l-intnetcore211 age=66210 idle=35 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=1010 addr=10.19.50.181:6379 laddr=10.19.50.184:37223 fd=89 name= age=1560455 idle=4 flags=M db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20520 events=r cmd=zincrby user=(superuser) redir=-1
id=5329715 addr=10.19.35.22:39498 laddr=10.19.50.184:6380 fd=41 name=catalog-api-l-intnetcore202 age=66156 idle=52 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5330446 addr=10.19.35.36:54056 laddr=10.19.50.184:6380 fd=28 name=catalog-admin-api-l-intnetcore216 age=65955 idle=15 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5330447 addr=10.19.35.36:54072 laddr=10.19.50.184:6380 fd=29 name=catalog-admin-api-l-intnetcore216 age=65955 idle=15 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=5329716 addr=10.19.35.22:39504 laddr=10.19.50.184:6380 fd=42 name=catalog-api-l-intnetcore202 age=66156 idle=52 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=16490 addr=10.19.50.183:55204 laddr=10.19.50.184:6380 fd=84 name= age=1556135 idle=1556135 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=auth user=default redir=-1
id=3762 addr=10.19.50.181:50474 laddr=10.19.50.184:6380 fd=79 name= age=1559694 idle=30 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=_FT._LIST user=default redir=-1
id=5330011 addr=10.19.35.34:55594 laddr=10.19.50.184:6380 fd=50 name=catalog-admin-api-l-intnetcore214 age=66077 idle=2 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5330012 addr=10.19.35.34:55606 laddr=10.19.50.184:6380 fd=53 name=catalog-admin-api-l-intnetcore214 age=66077 idle=2 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=7994 addr=10.19.50.185:45776 laddr=10.19.50.184:6380 fd=38 name= age=1558517 idle=1558517 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=auth user=default redir=-1
id=16489 addr=10.19.50.183:55196 laddr=10.19.50.184:6380 fd=74 name= age=1556136 idle=1556136 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=auth user=default redir=-1
id=5329407 addr=10.19.35.25:35774 laddr=10.19.50.184:6380 fd=31 name=catalog-api-l-intnetcore205 age=66240 idle=38 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5329408 addr=10.19.35.25:35778 laddr=10.19.50.184:6380 fd=39 name=catalog-api-l-intnetcore205 age=66240 idle=38 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=5329518 addr=10.19.35.33:38716 laddr=10.19.50.184:6380 fd=51 name=catalog-admin-api-l-intnetcore213 age=66210 idle=35 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5329519 addr=10.19.35.33:38722 laddr=10.19.50.184:6380 fd=52 name=catalog-admin-api-l-intnetcore213 age=66210 idle=35 flags=P db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1
id=5329405 addr=10.19.35.21:34566 laddr=10.19.50.184:6380 fd=22 name=catalog-api-l-intnetcore201 age=66240 idle=60 flags=r db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20504 events=r cmd=info user=default redir=-1
id=5329406 addr=10.19.35.21:34574 laddr=10.19.50.184:6380 fd=30 name=catalog-api-l-intnetcore201 age=66240 idle=60 flags=Pr db=0 sub=1 psub=0 multi=-1 qbuf=0 qbuf-free=0 argv-mem=0 obl=0 oll=0 omem=0 tot-mem=20496 events=r cmd=ping user=default redir=-1

------ MODULES INFO OUTPUT ------
# ReJSON_trace
ReJSON_trace:   0: redis_module::base_info_func
   1: rejson::__info_func
   2: modulesCollectInfo
   3: logModulesInfo
   4: printCrashReport
   5: sigsegvHandler
   6: <unknown>
   7: <unknown>


# search_version
search_version:2.8.1
search_redis_version:6.2.14 - oss

# search_index
search_number_of_indexes:7

# search_fields_statistics
search_fields_numeric:Numeric=8,Sortable=4
search_fields_tag:Tag=21

# search_dialect_statistics
search_dialect_1:0
search_dialect_2:0
search_dialect_3:0

# search_runtime_configurations
search_concurrent_mode:OFF
search_enableGC:ON
search_minimal_term_prefix:2
search_maximal_prefix_expansions:200
search_query_timeout_ms:500
search_timeout_policy:fail
search_cursor_read_size:1000
search_cursor_max_idle_time:300000
search_max_doc_table_size:1000000
search_max_search_results:1000000
search_max_aggregate_results:-1
search_search_pool_size:20
search_index_pool_size:8
search_gc_scan_size:100
search_min_phonetic_term_length:3

------ FAST MEMORY TEST ------
1411:S 05 Jun 2024 02:00:31.275 # main thread terminated
1411:S 05 Jun 2024 02:00:31.275 # Bio thread for job type #0 terminated
1411:S 05 Jun 2024 02:00:31.275 # Bio thread for job type #1 terminated
1411:S 05 Jun 2024 02:00:31.275 # Bio thread for job type #2 terminated

Fast memory test PASSED, however your memory can still be broken. Please run a memory test for several hours if possible.

=== REDIS BUG REPORT END. Make sure to include from START to END. ===
```

</details>

**To Reproduce**
This happens intermittently so I'm not sure how to reproduce it

**Expected behavior**
redis cluster is able to support RediSearch indices w/o crashing

**Environment (please complete the following information):**
 - OS: Rocky Linux release 8.9 (Green Obsidian)
 - CPU model: 4x Intel(R) Xeon(R) Gold 6258R CPU @ 2.70GHz x86_64
 - Version: 6.2.14

**Additional context**
- 10-node redis cluster. 2 nodes per server
- modules: RediSearch (w/ coordinator) and redisJSON

**Comment by GuyAv46** [comment]
Hey @mkoser thanks for reaching out!
Where do you get the module from? Are you building it from source?
The crash report is hard to follow as there are almost no calls in the stack trace. Do you have the server logs prior to the crash?

**Comment by mkoser** [comment]
Thanks for the quick response @GuyAv46 
We build the module in-house.

Here are the relevant logs just prior to the crash event:
```
1411:S 05 Jun 2024 02:00:00.312 * <module> Scanning index SamplingProgram-1.330.0.0-6-5-2024-2_00_00_3057133_AM in background
1411:S 05 Jun 2024 02:00:01.070 * 10000 changes in 300 seconds. Saving...
1411:S 05 Jun 2024 02:00:01.195 * Background saving started by pid 3338573
```

**Comment by GuyAv46** [comment]
Thanks! Seems unrelated.
I’ll have to ask if you built your binary from a modified fork. Since the so file name is not the same as the one our scripts generate, my script for stitching the debug symbols back to translate the stack trace didn’t work. I’ll modify it on Sunday, but if you have your own code you’ll have to do it yourself as stitching our 2.8.1 rocky9 symbols won’t help.

2 more notes

1. Version 2.8 doesn’t support redis 6.2. Maybe 2.8.1 mostly works, but we added some API calls that are not available on redis versions lower than 7.2. This even may explain the crash. If you have to use redis 6.2 I suggest using RediSearch version 2.6
2. I don’t know how you build your binary, but for cluster support you just need to call `make COORD=1`. You can follow the pinned issue for trouble shooting

**Comment by mkoser** [comment]
I'll start with upgrading to 7.2 and see how things look afterward.

Thanks for the help

**Comment by GuyAv46** [comment]
Closing for now. Feel free to re-open if more help is needed, or open a new ticket
